pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -373,30 +402,41 @@ func (r *ReconcileToolchainStatus) sendToolchainStatusUnreadyNotification(logger
 	}
```

Comment:
Move it to the line 408 where you initialize the contentString with the actual value in case of unready status. We do not need to marshal the entire status in case of ready status. It's not used there.

Reply:
i don't understand what you are suggesting, i removed the function "sendToolchainStatusUnreadyNotification" to have one merged function sendToolchainStatusNotification.

Reply:
I'm suggesting to move the code from lines 397-402:
```
	toolchainStatus = toolchainStatus.DeepCopy()
	toolchainStatus.ManagedFields = nil // we don't need these managed fields in the notification
	statusYaml, err := yaml.Marshal(toolchainStatus)
	if err != nil {
		return err
	}
```
to:
```
	if status == unreadyStatus {
		toolchainStatus = toolchainStatus.DeepCopy()
		toolchainStatus.ManagedFields = nil // we don't need these managed fields in the notification
		statusYaml, err := yaml.Marshal(toolchainStatus)
		if err != nil {
			return err
		}
		contentString = "<div><pre><code>" + string(statusYaml) + "</code></pre></div>"
		subjectString = adminUnreadyNotificationSubject
	} else if status == restoredStatus {
		contentString = "<div><pre>ToolchainStatus is back to ready status.</pre></div>"
		subjectString = adminRestoredNotificationSubject
	} else {
		return fmt.Errorf("invalid ToolchainStatusNotification status type: %s", status)
	}
```
or even:
```
	switch status {
	case unreadyStatus:
		toolchainStatus = toolchainStatus.DeepCopy()
		toolchainStatus.ManagedFields = nil // we don't need these managed fields in the notification
		statusYaml, err := yaml.Marshal(toolchainStatus)
		if err != nil {
			return err
		}
		contentString = "<div><pre><code>" + string(statusYaml) + "</code></pre></div>"
		subjectString = adminUnreadyNotificationSubject
	case restoredStatus:
		contentString = "<div><pre>ToolchainStatus is back to ready status.</pre></div>"
		subjectString = adminRestoredNotificationSubject
	default:
		return fmt.Errorf("invalid ToolchainStatusNotification status type: %s", status)
	}
```
So, we don't marshal the status yaml when we don't need it.